src/Microsoft.Framework.Runtime/Project.cs

```
@@ -188,6 +190,21 @@ internal static Project GetProjectFromStream(Stream stream, string projectName,
                 }
             }
 
+            var fileVersion = Environment.GetEnvironmentVariable("DNX_ASSEMBLY_FILE_VERSION");
+            if (string.IsNullOrWhiteSpace(fileVersion))
+            {
+                fileVersion = project.Version.ToString();
+            }
+
+            try
+            {
+                project.AssemblyFileVersion = new Version(fileVersion);
+            }
+            catch(FormatException ex)
+            {
+                throw new FormatException("The assembly file version is invalid: " + fileVersion, ex);
```

Nah. Just use Version.Parse and let that throw.